discovery-provider/src/challenges/challenge.py

```diff
@@ -186,14 +186,18 @@ def process(self, session, event_type: str, event_metadatas: List[EventMetadata]
                 .group_by(UserChallenge.user_id)
             ).all()
             challenges_per_user = dict(all_user_challenges)
+            new_user_challenges_specifiers: Set[str] = set()
             for new_metadata in new_challenge_metadata:
-                completion_count = challenges_per_user.get(new_metadata["user_id"], 0)
+                completion_count = challenges_per_user.get(
+                    new_metadata["user_id"], 0
+                ) + len(new_user_challenges_specifiers)
```

Comment:
Does this new specifier set need to be by `user_id`  as well? Also, do we want to handle this here or should my updater check and remove the extra challenges?

Reply:
Aha, this is a nice catch! I don't think the logic of the fix is quite right here though - you're making a set of all the new user_challenges (deduped by specifier) we make and then adding to the count of the user challenge per user. I think what we want here is a map of user_id -> new user_challenge created count, because this needs to happen on a per user basis. But, this definitely is the right place to do it, since this affects all aggregate challenges.

Reply:
yeah that makes sense, that's kinda what I was getting at in my first q